complex/src/lib.rs

```diff
@@ -22,6 +22,8 @@ extern crate rustc_serialize;
 #[cfg(feature = "serde")]
 extern crate serde;
 
+pub mod scalar;
```

Comment:
It's good to implement this in a module, but that shouldn't be public.  Just `pub use scalar::Scalar` to expose it.